lib/matplotlib/font_manager.py

```diff
@@ -180,24 +184,46 @@ def win32InstalledFonts(directory=None, fontext='ttf'):
 
     items = set()
     for fontdir in MSFontDirectories:
+        # System fonts
         try:
             with winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, fontdir) as local:
                 for j in range(winreg.QueryInfoKey(local)[1]):
-                    key, direc, tp = winreg.EnumValue(local, j)
-                    if not isinstance(direc, str):
+                    # Usually, value contains the name of the font file.
+                    key, value, tp = winreg.EnumValue(local, j)
+                    if not isinstance(value, str):
                         continue
                     # Work around for https://bugs.python.org/issue25778, which
                     # is fixed in Py>=3.6.1.
-                    direc = direc.split("\0", 1)[0]
+                    value = value.split("\0", 1)[0]
+                    try:
+                        path = Path(directory, value).resolve()
+                    except RuntimeError:
+                        # Don't fail with invalid entries.
+                        continue
+                    if path.suffix.lower() in fontext:
+                        items.add(str(path))
+        except (OSError, MemoryError):
+            continue
+
+        # User fonts
+        try:
+            with winreg.OpenKey(winreg.HKEY_CURRENT_USER, fontdir) as local:
+                for j in range(winreg.QueryInfoKey(local)[1]):
+                    # Usually, value contains the absolute path to the font
+                    # file.
+                    key, value, tp = winreg.EnumValue(local, j)
+                    if not isinstance(value, str):
+                        continue
                     try:
-                        path = Path(directory, direc).resolve()
+                        path = Path(value).resolve()
```

Comment:
I think this should still prepend a sensible base `direc` (which gets dropped anyways if `value` is absolute) or perhaps better just skip values that are not absolute to start with; otherwise non-absolute values get resolved relative to cwd which is certainly not correct.